modules/components/tag-input.ts

```diff
@@ -701,6 +707,74 @@ export class TagInputComponent extends TagInputAccessor implements OnInit {
         }
     }
 
+
+    /**
+     * @name DRAG_N_DROP_KEY
+     * @type {string}
+     */
+    private static DRAG_N_DROP_KEY: string = "dragged tag";
+
+    /**
+     * @name onDragStarted
+     * @param event
+     * @param index
+     */
+    public onDragStarted(event: any, index: number): void {
+        if (!this.draggZone){
+            return;
+        }
+        let draggedElement = this.items[index];
+        const storedElement = {zone: this.draggZone, value: draggedElement};
+        event.dataTransfer.setData(TagInputComponent.DRAG_N_DROP_KEY, JSON.stringify(storedElement));
+        this.items.splice(index, 1);
```

Comment:
can we avoid mutation and redeclare by using `.filter()` (or actually you could use `getItemsWithout()`) defined in helpers/accessor.ts which extends this class